_posts/2024-03-20-strimzi-kraft-migration.md

```diff
@@ -0,0 +1,268 @@
+---
+layout: post
+title:  "Migrate your Strimzi-operated cluster from ZooKeeper to KRaft"
+date: 2024-03-20
+author: paolo_patierno
+---
+
+In the previous [blog post](https://strimzi.io/blog/2024/03/21/kraft-migration/) we explored how Apache ZooKeeper is used to store Apache Kafka clusters metadata and what are the current limitations.
+The Apache Kafka community introduced the new KRaft protocol which allows you to overcome these limitations and store metadata in Kafka itself, taking into account that the ZooKeeper support will be removed soon.
+For this reason, there is a need to migrate existing clusters from using ZooKeeper to KRaft for storing metadata.
+We also covered the overall Kafka metadata migration procedure.
+The migration procedure can be done without downtime, but it is not that straightforward because it requires manual configurations and several rolling updates.
+The good news is that if your cluster is powered by Strimzi, this process is much easier and can be accomplished simply by annotating your `Kafka` resource.
+In this blog post, we are going to show you how the Strimzi Cluster Operator provides a semi-automated process to run the migration easily, so that you don't need to worry about updating configurations and restarting cluster nodes.
+
+<!--more-->
+
+### Strimzi's migration support
+
+If you are using the Strimzi operator to run your current ZooKeeper-based cluster on Kubernetes, the migration process is semi-automated.
+It is managed by updating the `strimzi.io/kraft` annotation on the `Kafka` custom resource.
+This allows the internal migration service to transition smoothly across the different phases.
+The operator will apply the necessary configuration to the nodes and then initiate a rolling update.
+It is also in charge of checking the migration status by looking at the metrics to make sure it was done successfully.
+
+Before the Strimzi 0.40.0 release, the `strimzi.io/kraft` annotation on the `Kafka` custom resource was used to define a ZooKeeper-based cluster, by using the `disabled` value, or a KRaft-based cluster, by using the `enabled` value.
+
+This is still used even in Strimzi 0.40.0.
+But now it is possible to use two additional values for the `strimzi.io/kraft` annotation:
+
+* `migration` to start the migration process
+* `rollback` to revert the migration itself
+
+During migration or rollback, the `enabled` and `disabled` values are used to finalize the procedure by ending with a cluster in KRaft mode or reverting to the ZooKeeper mode.
+
+One of the main prerequisites for the migration is having the ZooKeeper-based cluster using the `KafkaNodePool` resource(s) to run the brokers.
```

Comment:
Should we mention the prerequisites for Strimzi version >= 0.40.0?

Reply:
Wdym? We are already referring to Strimzi 0.40.0 as mentioned some lines before. So now I am adding prerequisites since this version (even because migration was not there before). Or I just didn't get what you meant :-)

Reply:
Yes, I know you've referring to Strimzi 0.40.0, but you didn't say it's one of the prerequisites. Of course if you think that is fine, I'm also OK. Thanks.

Reply:
I changed the first sentence in the paragraph as ... `If you are using the Strimzi operator 0.40.0 release to run ..."
I can't say "greater than" because it's the only release we have out now with this feature. But it would be obvious until we'll remove migration in one year because ZooKeeper not supported anymore.

Reply:
Sounds good to me. Thanks.